centos-upgrade-first-stage.sh

```diff
@@ -64,8 +64,18 @@ echo "Installing basic packages"
 sudo yum -y groupinstall development
 sudo yum -y install git yum-utils
 
+if test $is_centos6 -eq 1; then
+    sudo yum -y install python-pip
+    sudo pip install awscli --upgrade
+else
+    sudo yum -y install python-pip python-wheel awscli
```

Comment:
yum update of awscli for centos7 will bring bit older version 1.11.133 as per https://pkgs.org/download/awscli, but pip install get us latest cli (1.11.185). Is there any reason why you are not using pip install of awscli for centos7?

Reply:
What @dougalb said below.  In general, we should always use the vendor-provided package over building our own.  There are a couple reasons.  First, updates are much cleaner.  Second, if a customer later wants to install a package that depends on the awscli package, all kinds of hell would break loose if we installed outside the vendor packaging system if there are any file conflicts.  Third, it can cause significant customer confusion over who owns the software.

We have to install awscli via pip for CentOS 6, because there is no package for it.  But CentOS 7 has a package, so that's the right answer.